### PR TITLE
Fix slack SSO with Google

### PR DESCRIPTION
**Pull request by sad270**
fix issue: https://github.com/getferdi/ferdi/issues/1350

### CODE REVIEW

File: recipes/slack/index.js [hunk 1]
```diff
@@ -1,3 +1,17 @@
...
 
-module.exports = Franz => Franz;
\ No newline at end of file
+module.exports = (Franz) =>
+    class Slack extends Franz {
+        modifyRequestHeaders() {
+            return [
+                {
+                    headers: {
+                        'user-agent': window.navigator.userAgent.replace(/(Ferdi|Electron)\/\S+( \([^)]+\))?/g, "").trim(),
```
Comment (vraravam) [inline recipes/slack/index.js:9]: @sad270 - this line should no longer be needed.  The user now has the ability to override the `user-agent` from within the preferences/settings screen in Ferdi. Please confirm.

**Review by vraravam — CHANGES_REQUESTED** [review]

File: recipes/slack/index.js (hunk 1 above)
Comment (vraravam) [reply recipes/slack/index.js:9]: I am logging into slack via an okta-based SSO - and I do not need this "fix"

Comment (sad270) [reply recipes/slack/index.js:9]: Hello,
I already try to change the UA from the service configuration but it's doesn't work
I have a slack with Google Sign in only, and my Google Account have 2FA, the login fail after the 2FA authentification (when I click on "Yes" on my Phone)
I explain all here https://github.com/getferdi/ferdi/issues/1350#issuecomment-851336929

I try:
* Override UA from the service config without changing the `index.js`
* Override UA from the service config and with only the `requestFilters` (line 12) in the `index.js`
* Only `requestFilters` (line 12) in the `index.js` with default UA
* Only change `modifyRequestHeaders` UA (line 9) without the `requestFilters` (line 12)

It doesn't work all of them failed after the 2FA

Note: In my test, I override UA in service config with: `Mozilla/5.0 (Macintosh; Intel Mac OS X 11_4_0) AppleWebKit/537.36 (KHTML, like Gecko)1.1 Chrome/91.0.4472.77 Safari/537.36`

**Comment by sad270** [comment]
I use the last Nightly 😎

```
Version: 5.6.0-nightly.88
Electron: 13.1.7
Chrome: 91.0.4472.124
Node.js: 14.16.0
Platform: macOS Catalina
Arch: x64
Build date: Sun Jul 18 2021 04:08:58 GMT+0200 (Central European Summer Time)
Git SHA: 8b87a1e2
Git branch: nightly
```

File: recipes/slack/index.js (hunk 1 above)
Comment (JustoHermano) [reply recipes/slack/index.js:9]: My company is using slack login via okta aswell and login is currently failing aswell even with the user agent above

**Comment by vraravam** [comment]
@sad270 - please review this PR, fix it and resubmit or we can close it?

**Comment by sad270** [comment]
Yes, sorry i forgot this. I will work on this PR this weekend.

**Comment by sad270** [comment]
This can be closed since https://github.com/getferdi/ferdi/pull/1991 is merged

**Pull request closed without merging**